tsl/src/continuous_aggs/create.c

```diff
@@ -1529,27 +1560,41 @@ mattablecolumninfo_addentry(MatTableColumnInfo *out, Node *input, int original_q
 			}
 			else
 			{
-				out->mat_groupcolname_list = lappend(out->mat_groupcolname_list, pstrdup(colname));
+				if (!*skip_adding && ((TargetEntry *) input)->ressortgroupref > 0)
+					out->mat_groupcolname_list =
+						lappend(out->mat_groupcolname_list, pstrdup(colname));
 			}
+
 			coltype = exprType((Node *) tle->expr);
 			coltypmod = exprTypmod((Node *) tle->expr);
 			colcollation = exprCollation((Node *) tle->expr);
 			col = makeColumnDef(colname, coltype, coltypmod, colcollation);
 			part_te = (TargetEntry *) copyObject(input);
-			/* need to project all the partial entries so that materialization table is filled */
-			part_te->resjunk = false;
+
+			/* keep original resjunk if finalized or not time bucket */
+			if (!finalized || timebkt_chk)
+			{
+				/*
+				 * Need to project all the partial entries so that
+				 * materialization table is filled
+				 */
+				part_te->resjunk = false;
+			}
+
```

Comment:
This check of the time-bucketing functions above:
```
			if (IsA(tle->expr, FuncExpr))
timebkt_chk = function_allowed_in_cagg_definition(((FuncExpr *) tle->expr)->funcid);
```
is not correct and has never been correct.

Some things wrong with it:
  - `time_partition_col` is the default name given to those columns. This assumes it's only one column using a time-bucketing function in the target list, which may not be the case.
  - it only checks top-level function expressions, so if the function is hidden within an expression it is never checked and no special handling will occur.

E.g. this statement potentially breaks the code in various places:

```
CREATE MATERIALIZED VIEW conditions_summary
    WITH (timescaledb.continuous, timescaledb.materialized_only=true)
    AS SELECT
        time_bucket('1 days', time) AT TIME ZONE 'UTC',
        time_bucket('1 days', time) AT TIME ZONE 'UTC' + '1 day',
        avg(temperature)
        FROM conditions
        GROUP BY device_id, time_bucket('1 days', time)  WITH NO DATA;
```

Reply:
@fabriziomello let me take care of it in another PR.

Reply:
@mfundul Ok. let's create another issue and then discuss more there